src/Api/Controllers/CollectionsController.cs

```diff
@@ -189,35 +194,35 @@ public async Task PutUsers(Guid orgId, Guid id, [FromBody] IEnumerable<Selection
     [HttpPost("{id}/delete")]
     public async Task Delete(Guid orgId, Guid id)
     {
-        if (!await CanDeleteCollectionAsync(orgId, id))
+        var collection = await GetCollectionAsync(id, orgId);
+        var result = await _authorizationService.AuthorizeAsync(User, collection, CollectionOperations.Delete);
+        if (!result.Succeeded)
         {
             throw new NotFoundException();
         }
 
-        var collection = await GetCollectionAsync(id, orgId);
         await _deleteCollectionCommand.DeleteAsync(collection);
     }
 
     [HttpDelete("")]
     [HttpPost("delete")]
     public async Task DeleteMany([FromBody] CollectionBulkDeleteRequestModel model)
     {
-        var orgId = new Guid(model.OrganizationId);
         var collectionIds = model.Ids.Select(i => new Guid(i));
-        if (!await _currentContext.DeleteAssignedCollections(orgId) && !await _currentContext.DeleteAnyCollection(orgId))
+        var collections = await _collectionRepository.GetManyByManyIdsAsync(collectionIds);
+
+        if (!collections.Any())
         {
-            throw new NotFoundException();
+            throw new BadRequestException("No collections found.");
```

Comment:
I think that you could make a case for using a `BadRequestException` because the resulting collections are based solely on the `ids` that were passed in, thus (technically) resulting in a bad request. But since we've agreed upon the handler handling null resource checks, a `NotFoundException` is also an arguable exception!